stl/inc/expected

```diff
@@ -0,0 +1,1035 @@
+        }
+    }
+
+    // clang-format off
+    // TRANSITION, LLVM-46269, destructor order is significant
+    ~expected() requires is_trivially_destructible_v<_Ty> && is_trivially_destructible_v<_Err> = default;
+    // clang-format on
+
+
+    // [expected.object.assign]
+    template <class _Uty>
+        requires is_nothrow_move_constructible_v<_Uty>
+    struct _GuardTy {
+        constexpr _GuardTy(_Uty* _Target_, _Uty* _Tmp_) noexcept : _Target(_Target_), _Tmp(_Tmp_) {}
+        constexpr ~_GuardTy() noexcept {
+            if (_Target) {
+                _STD construct_at(_Target, _STD move(*_Tmp));
+            }
+        }
+        _Uty* _Target;
+        _Uty* _Tmp;
+    };
+
+    template <class _First, class _Second, class... _Args>
+    static constexpr void _Reinit_expected(_First& _New_val, _Second& _Old_val, _Args&&... _Vals) noexcept(
+        is_nothrow_constructible_v<_First, _Args...> || is_nothrow_move_constructible_v<_First>) // strengthened
+    {
+        if constexpr (is_nothrow_constructible_v<_First, _Args...>) {
+            _STD destroy_at(_STD addressof(_Old_val));
+            _STD construct_at(_STD addressof(_New_val), _STD forward<_Args>(_Vals)...);
+        } else if constexpr (is_nothrow_move_constructible_v<_First>) {
+            _First _Tmp(_STD forward<_Args>(_Vals)...);
+            _STD destroy_at(_STD addressof(_Old_val));
+            _STD construct_at(_STD addressof(_New_val), _STD move(_Tmp));
+        } else {
+            _Second _Tmp(_STD move(_Old_val));
+            _STD destroy_at(_STD addressof(_Old_val));
+
+            _GuardTy<_Second> _Guard{_STD addressof(_Old_val), _STD addressof(_Tmp)};
+            _STD construct_at(_STD addressof(_New_val), _STD forward<_Args>(_Vals)...);
+            _Guard._Target = nullptr;
+        }
+    }
+
+    constexpr expected& operator=(const expected& _Other) noexcept(
+        is_nothrow_copy_constructible_v<_Ty>&& is_nothrow_copy_constructible_v<_Err> //
+            && is_nothrow_copy_assignable_v<_Ty>&& is_nothrow_copy_assignable_v<_Err>) // strengthened
+        requires is_copy_assignable_v<_Ty> && is_copy_constructible_v<_Ty> && is_copy_assignable_v<_Err> //
+        &&(is_nothrow_move_constructible_v<_Ty> || is_nothrow_move_constructible_v<_Err>) {
+        if (_Has_value && _Other._Has_value) {
+            _Value = _Other._Value;
+        } else if (_Has_value) {
+            _Reinit_expected(_Unexpected, _Value, _Other._Unexpected);
+        } else if (_Other._Has_value) {
+            _Reinit_expected(_Value, _Unexpected, _Other._Value);
+        } else {
+            _Unexpected = _Other._Unexpected;
+        }
+
+        _Has_value = _Other._Has_value;
+        return *this;
+    }
+
+    constexpr expected& operator=(expected&& _Other) noexcept(
+        is_nothrow_move_constructible_v<_Ty>&& is_nothrow_move_constructible_v<_Err> //
+            && is_nothrow_move_assignable_v<_Ty>&& is_nothrow_move_assignable_v<_Err>) // strengthened

```

Comment:
[expected.object.assign]/6 specifies this, so it is not strengthened.

(Yesterday I noticed a similar issue for `expected<void, E>` but didn't go back and check the primary template.)